### PR TITLE
[SQL] `az sql db`: Add `--preferred-enclave-type` argument

### PR DESCRIPTION
**Pull request by vidit-msft**
**Related command**

- az sql db create
- az sql db show

**Description**

- Adds support for PreferredEnclaveType parameter for Database CRUD operations. 

**Testing Guide**

- az sql db create --preferred-enclave-type "VBS"
- az sql db show

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Review by evelyn-ys — COMMENTED** [review]
The changes LGTM but the new recordings seem to use advanced API version. Do you upgrade the SDK locally? Do you mean to bump version for this argument change?

**Comment by vidit-msft** [comment]
@evelyn-ys Thanks for the review. Yes, for the new parameter I upgraded the SDK locally and then used the new api version 2022-05-01-preview API version. The failing tests worked fine locally, when I recorded them. Anything I need to do to fix the 3 failing checks?

**Comment by jsntcy** [comment]
@vidit-msft, Where did you get the new SDK with 2022-05-01-preview API version? I didn't see any new SDK released with that new API version in https://pypi.org/project/azure-mgmt-sql/#history.
If there is no public SDK with 2022-05-01-preview API version, you cannot support this feature in Azure CLI as Azure CLI depends on public SDK.

**Comment by vidit-msft** [comment]
@jsntcy I requested a private python SDK build [[resource-manager] Python: 'Release of Database Services - 2022-05-01' · Issue #3574 · Azure/sdk-release-request (github.com)](https://github.com/Azure/sdk-release-request/issues/3574#issuecomment-1360638213) and used it for updating azure cli commands and record tests. Can I request a public SDK release and then we can merge this PR?

**Comment by jsntcy** [comment]
Once you use the public SDK, you need test all SQL tests lively (There are many tests for SQL as many teams contribute to SQL commands) to avoid regression, if there is no issue, we can merge. 
I'm not sure if you have enough time to run those tests as you may need some time to get the public SDK. If you cannot catch this sprint, we probably postpone it to next sprint (one month later).

**Comment by vidit-msft** [comment]
@jsntcy I talked to the Python SDK team and they released a new python sdk https://pypi.org/project/azure-mgmt-sql/4.0.0b6/ which relies on 2022-05-01-preview API version. Should I bump up the azure-mgmt-sql version in Azure-CLI and re-record the tests now?

**Comment by evelyn-ys** [comment]
Sure pls bump SDK version if it's required for this feature.

We have an internal pipeline for bumping version, see https://github.com/Azure/azure-cli/blob/dev/doc/how_to_bump_SDK_version_in_cli.md

**Comment by vidit-msft** [comment]
@evelyn-ys I don't seem to have access to the Regression test pipeline mentioned in [azure-cli/how_to_bump_SDK_version_in_cli.md at dev · Azure/azure-cli (github.com)](https://github.com/Azure/azure-cli/blob/dev/doc/how_to_bump_SDK_version_in_cli.md). I have already requested access for it. When I bumped up the sdk version and tried to run all the sql tests, I see some failures i.e 29 failed and 45 tests passed. How do we make sure all tests pass with the bumped up sdk version to proceed forward?

image

**Comment by evelyn-ys** [comment]
Did you rerun all tests with `--live` and then saw the 29 failure tests? You need to figure out why they failed and fix one by one.

**Comment by jsntcy** [comment]
You need contact other teams to rerun those failed tests lively. (The root cause is that some resources in tests are hardcoded instead of creating dynamically)

File: src/azure-cli/azure/cli/command_modules/sql/tests/latest/recordings/test_metric_alert_for_sql_database_scope.yaml [hunk 1]
```diff
@@ -0,0 +1,920 @@
+interactions:
```
Comment (ostojicb) [inline src/azure-cli/azure/cli/command_modules/sql/tests/latest/recordings/test_metric_alert_for_sql_database_scope.yaml:1]: this recording should be on a different path, here:
`https://github.com/Azure/azure-cli/blob/main/src/azure-cli/azure/cli/command_modules/monitor/tests/latest/recordings/test_metric_alert_for_sql_database_scope.yaml`

Comment (vidit-msft) [reply src/azure-cli/azure/cli/command_modules/sql/tests/latest/recordings/test_metric_alert_for_sql_database_scope.yaml]: oh, my bad. Fixed it.

**Review by ostojicb — APPROVED** [review]

**Comment by vidit-msft** [comment]
@evelyn-ys All the checks pass now, can we please merge this PR and have it be part of the Azure-CLI January release?

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**